Source/JavaScriptCore/ftl/FTLLowerDFGToB3.cpp

```diff
@@ -13971,6 +13974,42 @@ IGNORE_CLANG_WARNINGS_END
         }
     }
 
+    void compileNumberIsFinite()
+    {
+        switch (m_node->child1().useKind()) {
+        case DoubleRepUse: {
+            LValue argument = lowDouble(m_node->child1());
+            LValue negArgument = m_out.neg(argument);
+            LValue result = m_out.doubleAdd(argument, negArgument);
+            setBoolean(m_out.equal(result, result));
```

Comment:
Is it correct? (It is `equal`, not `doubleEqual`). Also can you add tests for them?